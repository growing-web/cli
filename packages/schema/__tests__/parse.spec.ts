import { test, describe, expect } from 'vitest'
import { parse } from '@growing-web/web-builder-schema'
import { resolve } from 'pathe'

describe('parse manifest test.', () => {
  test('wrong configuration value.', async () => {
    const root = resolve(__dirname, 'fixtures/wrong-configuration/')
    let err: any
    try {
      await parse({ root, manifestFileName: 'project-manifest.json' })
    } catch (error) {
      err = error
    }
    expect(() => {
      throw err.toString().replace(process.cwd(), '')
    }).toThrowErrorMatchingSnapshot()
  })

  test('correct environment variable configuration.', async () => {
    const root = resolve(__dirname, 'fixtures/env/')

    const manifest = await parse({
      root,
      manifestFileName: 'project-manifest.json',
      mode: 'production',
    })
    expect(manifest.publicPath).toBe('http://dev.test.com')
    expect(manifest.server?.port).toBe(3200)
    expect(manifest.server?.host).toBe('http://dev.test.com')
  })
})
