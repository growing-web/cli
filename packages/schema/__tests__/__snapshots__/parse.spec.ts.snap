// Vitest Snapshot v1

exports[`parse manifest test. > wrong configuration value. 1`] = `
"ValidationError: Invalid Manifest object. Project has been initialized using a Manifest object that does not match the API schema.
 - Manifest.entry should be a string."
`;
